lib/usecases/login_use_case.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/api/repository/login_repository.dart';
+import 'package:flutter_survey/api/request/login_request.dart';
+import 'package:flutter_survey/usecases/base/base_use_case.dart';
+import 'package:injectable/injectable.dart';
+import 'package:flutter_survey/model/login_model.dart';
+
+@Injectable()
+class LoginUseCase extends UseCase<LoginModel, LoginRequest> {
+  final LoginRepository _loginRepository;
+
+  const LoginUseCase(this._loginRepository);
+
+  @override
+  Future<Result<LoginModel>> call(LoginRequest input) {
```

Comment:
```suggestion
  Future<Result<LoginModel>> call(LoginInput input) {
```

Should we create a `LoginInput` class, to be more specific on what's being inputted from the UI layer?
```
LoginInput({
  required this.email,
  required this.password,
});
```

Inside `login_repository.dart`, we create the `LoginRequest` object, so it's specific to the network layer:

```
LoginRequest loginRequest = LoginRequest(
  grantType: "password",
  email: email,
  password: password,
  clientId: Env.restApiClientId,
  clientSecret: Env.restApiClientSecret,
);

final response = await _loginService.doLogin(request);
```

Reply:
@Tuubz I did exactly this, but it creates problem when I wrote test for usecase class. In this style `Env.restApiClientId` and `Env.restApiClientSecret` cannot be read when try to read from test method call.

Reply:
@Shayokh144 Would this be able to resolve the issue? 🙇 

```
  FlutterConfig.loadValueForTesting({
    'CLIENT_ID': 'CLIENT_ID',
    'CLIENT_SECRET': 'CLIENT_SECRET',
  });
```

**Reference**: https://github.com/chornerman/survey-flutter/blob/develop/test/api/repository/auth_repository_test.dart

Reply:
@Tuubz thanks, it works. I updated the code.